hr_announcement/models/announcement.py

```diff
@@ -0,0 +1,106 @@
+# Copyright 2023 Tecnativa - Pilar Vargas
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import api, fields, models
+
+
+class Announcement(models.Model):
+    _inherit = "announcement"
+
+    announcement_type = fields.Selection(
+        selection_add=[
+            ("employee", "Employee"),
+            ("department", "Department"),
+            ("job_position", "Job Position"),
+        ],
+        ondelete={
+            "employee": "set default",
+            "department": "set default",
+            "job_position": "set default",
+        },
+    )
+
+    employee_ids = fields.Many2many(
+        comodel_name="hr.employee",
+        relation="hr_employee_announcements",
+        column1="announcement",
+        column2="employee",
+        string="Employees",
+    )
+    department_ids = fields.Many2many(
+        comodel_name="hr.department",
+        relation="hr_department_announcements",
+        column1="announcement",
+        column2="department",
+        string="Departments",
+    )
+    position_ids = fields.Many2many(
+        comodel_name="hr.job",
+        relation="hr_job_position_announcements",
+        column1="announcement",
+        column2="job_position",
+        string="Job Positions",
+    )
+
+    @api.depends(
+        "specific_user_ids",
+        "user_group_ids",
+        "employee_ids",
+        "department_ids",
+        "position_ids",
+    )
+    def _compute_allowed_user_ids(self):
+        res = super()._compute_allowed_user_ids()
```

Comment:
Are you sure super is excluding your new types?: https://github.com/OCA/server-ux/blob/8311de64e311120a758db38cabf8993bb86907cf/announcement/models/announcement.py#L83